c++/src/Vector.cc

```diff
@@ -88,6 +88,20 @@ namespace orc {
     }
   }
 
+  void EncodedStringVectorBatch::decodeDictionary() {
+    if (dictionaryDecoded) return;
+
+    size_t n = index.size();
+    resize(n);
+
+    for (size_t i = 0; i < n; ++i) {
+      if (!hasNulls || notNull[i]) {
+        dictionary->getValueByIndex(index[i], data[i], length[i]);
+      }
+    }
+    dictionaryDecoded = true;
```

Comment:
```suggestion
```

Reply:
dictionaryDecoded had been moved to `ColumnVectorBatch`